src/material-experimental/mdc-slider/slider-thumb.ts

```diff
@@ -87,10 +105,44 @@ export interface MatSliderDragEvent {
   /** Indicates which slider thumb this input corresponds to. */
   thumb: Thumb;
 
+  /** Whether the value of this slider thumb input has been set. */
+  _initialized: boolean = false;
+
   constructor(
     @Inject(DOCUMENT) private readonly _document: Document,
     readonly _elementRef: ElementRef<HTMLInputElement>,
-    ) {}
+    private readonly _slider: MatSlider,
+    ) {
+      // Initializing the min and max in the constructor guarantees that they will be
+      // defined by the time the value gets set. If the range is not defined before we
+      // try to set the value, we can run into the issue where the value is outside of
+      // the default range and get capped to the default min or max.
+      this._elementRef.nativeElement.min = this._slider.min.toString();
+      this._elementRef.nativeElement.max = this._slider.max.toString();
```

Comment:
Could you clarify why the min/max needs to be set _in the DOM_ at this point? I would have thought that having the internal state set before the `value` is set should be sufficient.